### ISSUE TITLE
FP2 enters airplane mode in the middle of calls

### ISSUE THREAD

**Issue by leebaylis** [ubports/ubuntu-touch#1316]
Hi, 

I was having this problem on and off for a while but it has got much worse since the recent major upgrade. I'm now running 16.04 LTS - armhf (20191121).

I have figured out that what's happening is that when I am making a call, my ear is pulling down the top menu and pressing things on the screen. 

Previously, the menu pulled down on whichever sub item (e.g. notifications) I had used last. 

Since the upgrade, the menu seems to be trying to pull down on a specific sub item depending on whereabouts my finger (or ear) is.

On my FP2, since the last upgrade most of the time none of the locations at the top of the screen work anymore to pull down the menu except the one 2/3 along from the left corresponding to Network settings. 

The next issue is that you can immediately start to interact with the options on the network screen before the menu has been pulled all the way down. 

So, in summary, what is happening is that in the middle of a call, my ear is touching the screen, pulling down into the networking menu, and pressing the flight mode option. 

My calls then terminate. 

I think before it was sporadic because it only used to happen when the last item I had used was networking. But now it happens pretty much every time I make a call and it is starting to drive me crazy. 

Please help! Maybe the menus could be stopped from being interactive until the bar is pulled all the way down? Or maybe there could be an option to lock the pull down during calls?

Thanks,
Lee

### Illustrations

- Device: FP2
- Channel: Development
- Build: 16.04 LTS - armhf (20191121)

**Comment by Flohack74** [comment]
The problem is probably the priximity sensor. I have similar problems on the Oneplus One from time to time. Can you reboot, then make a call and try to cover the top part of the smartphone with your hand? Does the display go black?

**Comment by leebaylis** [comment]
Thanks for getting back so quickly! I just tested calling and put my finger over the top, and the screen went black as you say. 

I didn't realise there was a proximity sensor! Learn something new every day :)

So maybe my ear does not register as blocking the proximity sensor :(

**Comment by Flohack74** [comment]
Well no, your ear does well ^^ try to get the point of the sensor closer to your ear maybe. But, as i said, we sometimes have fa failing check and then the phone does not measure at all. Also this happens when you have the phone in your pocket, and you get notifications, calls etc... weird things can happen ^^

**Comment by mardy** [comment]
And can you try pressing the screen while it goes black during a call (like in the previous experiment)? Maybe the screen goes black, but the touchscreen is still registering events?

**Comment by UniversalSuperBox** [comment]
According to https://github.com/ubports/ubuntu-touch/issues/1250, this is indeed the case.

**Comment by leebaylis** [comment]
OK some more testing!
1) when I put my finger over the proximity sensor and make a call, I cannot interact with the screen when it goes black
2) Now that I know about the proximity sensor, I have been looking at the screen out of the corner of my eye during calls and I notice that the black is flickering on and off -- so definitely my ear and head are not causing the proximity sensor to trigger.

**Comment by UniversalSuperBox** [comment]
Hi @leebaylis, have you continued to experience this problem?

It isn't a perfect solution, but the default Android image for the Fairphone 2 includes a utility to calibrate the proximity sensor. This may help the sensor's accuracy during phone calls.

**Comment by leebaylis** [comment]
Hi guys, 
Thanks for all your help so far -- My touchscreen has actually now just gone completely berserk, first it it wouldn't register presses at all in a strip down the left hand side and now it is inventing its own presses when I am not touching it. I am probably going to replace the module, which may fix this issue too -- the upgrade may have been a coincidence or a red herring. So please feel free to put this issue on hold and I'll update you when I get a new touchscreen. 
Thanks!

**Comment by UniversalSuperBox** [comment]
Hi @leebaylis,

It's been a long time since this issue last received a reply, are you still having this problem with your Fairphone 2?

**Comment by leebaylis** [comment]
Thanks for checking -- I got a new touchscreen and it's not going into airplane mode anymore -- but now that I know about the proximity sensor I have kept on looking out of the corner of my eye and I was still seeing the screen come on and off. 

But I guess my issue is resolved so I have closed! Thanks for the help!

**Comment by Flohack74** [comment]
Thanks!
